redfish-core/lib/processor.hpp

```diff
@@ -48,6 +48,55 @@ constexpr std::array<std::string_view, 2> processorInterfaces = {
     "xyz.openbmc_project.Inventory.Item.Cpu",
     "xyz.openbmc_project.Inventory.Item.Accelerator"};
 
+/**
+ * @brief Workaround to handle DCM (Dual-Chip Module) package for Redfish
+ *
+ * Make sure processor modeled as dual chip module ("dcmN-cpuN"),
+ * If yes then, replace Redfish processor id as "dcmN/cpuN" and check with
+ * given object path because Redfish does not support chip module concept.
+ *
+ * @param[in] processorId - The Redfish processor Id
+ * @param[in] objectPath  - The D-Bus object path that contain the processor
+ *                          segment
+ *
+ * @return true if matched with the given object path else false.
+ *
+ * @note Inventory modeled as "dcmN/cpuN" to support DCM so wherever using
+ *       Redfish processor id as "dcmN-cpuN" then this function (it support
+ *       both SCM and DCM) can be used for the inventory processor object
+ *       path validation.
+ */
+inline bool
+    isProcObjectMatched(const std::string& processorId,
+                        const sdbusplus::message::object_path& objectPath)
+{
+    bool isMatched = false;
+    if (processorId.find("dcm") != std::string::npos)
+    {
+        std::size_t delimiterPos = processorId.find('-');
+        if (delimiterPos != std::string::npos)
+        {
+            std::string procParent = processorId.substr(0, delimiterPos);
+            std::string procId = processorId.substr(delimiterPos + 1,
+                                                    processorId.length());
```

Comment:
I think this is tokenizing processorId by delimitting '-'.
It can be done using `bmcweb::split()` (which could be more readable).

```
     std::vector<std::string> substrs;
     bmcweb::split(substrs, processorId, '-');

     if(substrs.size() == 2)
     {
               const std::string& procParent = substrs[0];
              const std::string& procId = substrs[1];
             ...
```

Reply:
Do you mean boost::split() here?
I can't find a bmcweb::split().

Reply:
I agree that is slightly easier to read. However it is not exactly the same thing semantically. 

With the current code if there are multiple `'-'` characters in the `processorId `then it will be handled with the split always into two elements. It will then do the following check to see if the names match or not.

But using bmcweb::split() a similar string would be split into three or more elements. Then it would fall out to the `!isMatched` case unless extra handling was put in to check for that case. The extra handling would make this harder to read than the original code.

I don't think it is likely there would be multiple `'-'` characters, but I'm hesitant to change the semantics.

Is that acceptable to you?

Reply:
Wouldn't this be correct only if it forms  like `"dcmN-cpuN"` (with it starts with "dcm")?
Or, if  the logic needs to allow more than 2 "-", it can be handled without checking 'substrs.size() == 2".

I'm okay to keep the original logic.